app/src/main/java/com/eis/phoneringer/structure/AppManager.java

```diff
@@ -0,0 +1,98 @@
+package com.eis.phoneringer.structure;
+
+import android.content.Context;
+import android.media.Ringtone;
+import android.os.Build;
+import android.os.Handler;
+import android.os.Looper;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.eis.smslibrary.SMSHandler;
+import com.eis.smslibrary.exceptions.InvalidSMSMessageException;
+import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+
+/**
+ * This is a singleton class used to manage a received RingCommand or to send one
+ *
+ * @author Alberto Ursino, Luca Crema, Alessandra Tonin, Marco Mariotto
+ */
+public class AppManager {
+
+    private static final String WRONG_PASSWORD_TOAST = "Wrong password";
+    private static final int TIMEOUT_TIME = 30 * 1000; //30 seconds
+
+    /**
+     * Instance of the class that is instantiated in getInstance method
+     */
+    private static AppManager instance = null;
+
+    /**
+     * Private constructor
+     */
+    private AppManager() {
+    }
+
+    /**
+     * @return the AppManager instance
+     */
+    public static AppManager getInstance() {
+        if (instance == null)
+            instance = new AppManager();
+        return instance;
+    }
+
+    /**
+     * If the password of the RingCommand received is valid then play ringtone for fixed amount of time
+     *
+     * @param context     of the application
+     * @param ringCommand received
+     * @param ringtone    to be played
+     */
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    public void onRingCommandReceived(Context context, @NonNull RingCommand ringCommand, final Ringtone ringtone) {
+        final RingtoneHandler ringtoneHandler = RingtoneHandler.getInstance();
```

Comment:
Class RingtoneHandler does not exist in the [branch](https://github.com/AlbertoUrsino/killer-app/tree/AppManager) you're trying to push from, how could you check if your code even compiles?

Reply:
I simply add all the necessary files from the outside and then only commit the class that has to go to pull request.
All these files will be stashed in my local repo.

Reply:
Of course the files I use are:

- https://github.com/Cogno-IDU/killer-app
- https://github.com/Cogno-IDU/smslibrary

Reply:
Why don't you just pull from Cogno-IDU/killer-app/master to AlbertoUrsino/killer-app/master, so that you get a functioning app in your branch?
I don't exactly understand why you used git stash to put aside the "outsider" files needed to have a working app: from what I understand stashing is used when you're working on a branch and you have to switch to another, but you don't want to lose the changes you made to your working directory and you also don't want to commit them.

Reply:
If I merge all the "outsider" files in my master and then call a pull request.. Well, I tried but in the pull request will being show all the "ousiders" files.. I don't know how to resolve this problem, I tried with "cherry-commits" but I didn't understand them so I opted for this method and it works.

Reply:
> Well, I tried but in the pull request will being show all the "ousiders" files as "changed files"
Could you elaborate on this? From what branch and onto what branch did you pull (let's use the terms "origin" to refer to your GitHub fork of killer-app, and "upstream" to refer to Cogno-IDU's killer-app)?
Maybe upload on YouTube a video of what happens in Android Studio when you merge upstream/master into your local branch AppManager, then push to origin/AppManager, then make a pull request from AlbertoUrsino:AppManager to Cogno-IDU:master?

Reply:
What should happen is that you update the submodule from the commit it's on to a new commit based on the central repo. Since you're updating these you'll see the smslibrary submodule files as changed (since they in fact are changed). This is not a problem if they get in the pull request, you just have to update us that you updated the submodule, so we know that any file related to smslibrary is already ok since we reviewed it in it's own repository